lib/resty/radixtree.lua

```diff
@@ -210,6 +223,132 @@ local function insert_tab_in_order(tab, val, func)
     insert_tab(tab, val)
 end
 
+local function update_tab_in_order(tab, val, func)
```

Comment:
Why not use table.sort directly?

Example:

```
> t1={a=3,b="foo"}
> t2={a=1,b="bar"}
> t3={a=2,b="?"}
> t={t3,t2,t1}
> table.sort(t, function(tt1,tt2) return tt1.a < tt2.a end)
> =t
table: 0x7f15682c0cd0
> =require("inspect")(t)
{ {
    a = 1,
    b = "bar"
  }, {
    a = 2,
    b = "?"
  }, {
    a = 3,
    b = "foo"
  } }
>

```

Reply:
update_tab_in_order() is efficient than update and table.sort.  Table.sort uses auxsort() c++ quicksort function which recursively compares and swaps elements many times. But update_tab_in_order() only compare table length times, insert new item, remove old item and move latter items plus 1 index.   And it runs well in our productive environment.

Reply:
Ok, then could you supply two test cases to verify it works like table.sort? Thanks.